hoodie-cli/src/main/java/com/uber/hoodie/cli/commands/HoodieLogFileCommand.java

```diff
@@ -0,0 +1,278 @@
+  public String showLogFileCommits(
+      @CliOption(key = "logFilePathPattern", mandatory = true, help = "Fully qualified path for the log file")
+      final String logFilePathPattern) throws IOException {
+
+
+    List<String> logFilePaths = Arrays.stream(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+        HoodieCLI.tableMetadata.getHadoopConf()).globStatus(new Path(logFilePathPattern)))
+        .map(status -> status.getPath().toString()).collect(Collectors.toList());
+    Map<String, Integer> commitToCount = Maps.newHashMap();
+
+    for(String logFilePath: logFilePaths) {
+      FileStatus[] fsStatus = FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+          HoodieCLI.tableMetadata.getHadoopConf()).listStatus(
+          new Path(logFilePath));
+      Schema writerSchema = new AvroSchemaConverter().convert(readSchemaFromLogFile(new Path(logFilePath)));
+      HoodieLogFormat.Reader reader = HoodieLogFormat.newReader(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
+          HoodieCLI.tableMetadata.getHadoopConf()),
+          new HoodieLogFile(fsStatus[0].getPath()), writerSchema, true);
+
+      // read the avro blocks
+      while (reader.hasNext()) {
+        HoodieLogBlock n = reader.next();
+        String commitTime;
+        if (n instanceof HoodieCorruptBlock) {
+          try {
+            commitTime = n.getLogMetadata().get(HoodieLogBlock.LogMetadataType.INSTANT_TIME);
+          } catch (Exception e) {
+            commitTime = "corrupt_block_1";
+            // could not read metadata for corrupt block
+          }
+        } else {
+          commitTime = n.getLogMetadata().get(HoodieLogBlock.LogMetadataType.INSTANT_TIME);
+        }
+        int recordCount = 0;
+        if (n instanceof HoodieAvroDataBlock) {
+          recordCount = ((HoodieAvroDataBlock) n).getRecords().size();
+        }
+        if (commitToCount.containsKey(commitTime)) {
+          commitToCount.put(commitTime, recordCount + commitToCount.get(commitTime));
+        } else {
+          commitToCount.put(commitTime, recordCount);
+        }
+      }
+    }
+    String[][] rows = new String[commitToCount.size() + 1][];
+    int i = 0;
+    for(Map.Entry<String, Integer> entry : commitToCount.entrySet()) {
+      String [] commits = new String[2];
+      commits[0] = entry.getKey().toString();
+      commits[1] = String.valueOf(entry.getValue());
+      rows[i] = commits;
+      i++;
+    }
+    return HoodiePrintHelper.print(
+        new String[]{"CommitTime", "RecordCount"}, rows);
+  }
+
+  @CliCommand(value = "show logfile metadata", help = "Read details from log files")
+  public String showLogFileDetails(
+      @CliOption(key = "logFilePathPattern", mandatory = true, help = "Fully qualified path for the log file")
+      final String logFilePathPattern) throws IOException {
+
+
+    List<String> logFilePaths = Arrays.stream(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
```

Comment:
could you think about some ways of sharing code between these methods? seems like a lot of code is repeated here..

Reply:
yeah, did that now